src/main/kotlin/org/rust/ide/notifications/RsNotificationProvider.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.notifications
+
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.EditorNotificationPanel
+import com.intellij.ui.EditorNotifications
+
+abstract class RsNotificationProvider(
+    protected val project: Project
+) : EditorNotifications.Provider<EditorNotificationPanel>() {
+
+    protected abstract val disablingKey: String
```

Comment:
Consider moving it to the constructor. (and remove redundant `const`s)

Reply:
I've rewritten implementation as we had discussed in person.
Now it's possible to provide `disablingKey` per file. So I'd prefer not to pass any constant into the constructor to keep base implementation the most flexible